app/components/sales-contact-form.js

```diff
@@ -0,0 +1,78 @@
+import Component from '@ember/component';
+import { bool, reads } from '@ember/object/computed';
+import { task } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+
+export default Component.extend({
+  tagName: '',
+
+  flashes: service(),
+  raven: service(),
+  store: service(),
+
+  requiredMark: 'Required',
+
+  isSubmitting: reads('send.isRunning'),
+  isSuccess: bool('send.lastSuccessful.value'),
+
+  utmSource: 'travis-web',
+
+  send: task(function* () {
+    const { name, email, teamSize, phone, message, utmSource } = this;
+    const data = {
+      name,
+      email,
+      team_size: teamSize,
+      phone,
+      message,
+      utm_source: utmSource,
+    };
+
+    try {
+      const lead = this.store.createRecord('lead', data);
+
+      yield lead.save();
+      lead.unloadRecord();
+      this.reset();
+
+      return true;
+    } catch (error) {
+      this.flashes.error(
+        "Something went wrong while submitting your request. We're working to fix it!"
+      );
+      this.raven.logException('Lead/Close request: API request error');
+      throw error;
+    }
+  }).drop(),
+
+  reset() {
+    this.setProperties({
+      name: '',
+      email: '',
+      teamSize: '',
+      phone: '',
+      message: '',
+    });
```

Comment:
I'd suggest another improvement. You could have `lead` record directly on component instance:
```js
export default Component.extend({
  lead: null
});
```
In that case `reset` would be just:
```js
reset() {
  if (this.lead) this.lead.unloadRecord();
  this.lead = this.store.createRecord('lead');
}
```
This would also allow to bind lead properties directly in template:
```hbs
<form.field
  @value={{this.lead.name}}
  @onChange={{action (mut this.lead.name)}}
>
```
And then the `send` would be as simple as:
```js
send: task(function* () {
  try {
    yield this.lead.save();
    this.reset();
    return true;
  } catch (error) {
    ...
  }
})
```

How does it sound to you?

Reply:
ahh neat, this sounds good to me. Change made :smile: